src/main.tsx

```diff
@@ -87,30 +87,47 @@ function setupSharedWorkerAndRepo() {
 
 const repo = setupSharedWorkerAndRepo()
 
-// TODO: Yikes on bikes, ought to redesign this
-const findOrMakeDoc = async (key: string): Promise<DocumentId> => {
-  let docId = new URLSearchParams(window.location.search).get(key)
+async function findOrMakeDeviceDoc(): Promise<DocumentId> {
+  const deviceDocId = await localforage.getItem("deviceDocId")
 
-  if (!docId) {
-    docId = await localforage.getItem(key)
+  if (deviceDocId) {
+    return deviceDocId as DocumentId
   }
-  if (!docId) {
-    const handle = repo.create()
-    docId = handle.documentId
-    if (key == "workspaceDocId") {
-      createWorkspace({}, handle)
-    }
-    if (key == "deviceDocId") {
-      createDevice({}, handle)
-    }
-    await localforage.setItem(key, docId)
+
+  const handle = repo.create()
+  const newDeviceDocId = handle.documentId
+
+  createDevice({}, handle)
+
+  await localforage.setItem("deviceDocId", newDeviceDocId)
+
+  return newDeviceDocId
+}
+
+async function findOrMakeProfileDoc(): Promise<DocumentId> {
+  // profiles used to be called workspaces, for backwards compatibility keep the old name here
```

Comment:
let's just do a one-off migration for this and be done with it -- update the key name and put a date in a readme (end of january?) when we remove support for the old key